src/gt4py/eve/concepts.py

```diff
@@ -58,58 +57,42 @@ class SymbolRef(ConstrainedStr, regex=_SYMBOL_NAME_RE):
 
 @datamodels.datamodel(slots=True, frozen=True)
 class SourceLocation:
-    """Source code location (line, column, source)."""
+    """File-line-column information for source code."""
 
+    filename: Optional[str]
     line: int = datamodels.field(validator=_validators.ge(1))
     column: int = datamodels.field(validator=_validators.ge(1))
-    source: str
     end_line: Optional[int] = datamodels.field(validator=_validators.optional(_validators.ge(1)))
     end_column: Optional[int] = datamodels.field(validator=_validators.optional(_validators.ge(1)))
 
-    @classmethod
-    def from_AST(cls, ast_node: ast.AST, source: Optional[str] = None) -> SourceLocation:
-        if (
-            not isinstance(ast_node, ast.AST)
-            or getattr(ast_node, "lineno", None) is None
-            or getattr(ast_node, "col_offset", None) is None
-        ):
-            raise ValueError(
-                f"Passed AST node '{ast_node}' does not contain a valid source location."
-            )
-        if source is None:
-            source = f"<ast.{type(ast_node).__name__} at 0x{id(ast_node):x}>"
-        return cls(
-            ast_node.lineno,
-            ast_node.col_offset + 1,
-            source,
-            end_line=ast_node.end_lineno,
-            end_column=ast_node.end_col_offset + 1 if ast_node.end_col_offset is not None else None,
-        )
```

Comment:
Just a question, has this been moved to somewhere else or removed? Why?

Reply:
I moved the functionality to the frontend: https://github.com/GridTools/gt4py/blob/89fc3a0a051cf4c1d85143c8a1351b32375b4c5a/src/gt4py/next/ffront/dialect_parser.py#L104

I wanted to break down the dependency between `SourceLocation` and the `ast` module to respect the Single Responsibility Principe so that `SourceLocation` is not taking up the responsibility of converting between different representations of location information.

I would be perfectly fine with a free function outside the frontend that converts between `ast` and `SourceLocation`, but I didn't see the need for it yet.